src/github.com/matrix-org/dendrite/clientapi/routing/room_tagging.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/accounts"
+	"github.com/matrix-org/dendrite/clientapi/httputil"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/dendrite/common"
+	"github.com/matrix-org/gomatrixserverlib"
+	"github.com/matrix-org/util"
+)
+
+// NewMTag creates and returns a new MTag type variable
+func NewMTag() common.MTag {
+	return common.MTag{
+		Tags: make(map[string]common.TagProperties),
+	}
+}
+
+// GetTag implements GET /_matrix/client/r0/user/{userID}/rooms/{roomID}/tags
+func GetTag(req *http.Request, accountDB *accounts.Database, userID string, roomID string) util.JSONResponse {
+
+	if req.Method != http.MethodGet {
+		return util.JSONResponse{
+			Code: http.StatusMethodNotAllowed,
+			JSON: jsonerror.NotFound("Bad method"),
+		}
+	}
+
+	mtag := NewMTag()
+
+	localpart, _, err := gomatrixserverlib.SplitID('@', userID)
+	if err != nil {
+		return httputil.LogThenError(req, err)
+	}
+
+	data, err := accountDB.GetAccountDataByType(
+		req.Context(), localpart, roomID, "m.tag",
```

Comment:
We should add "m.tag" to gomatrixserverlib to avoid raw strings.
/ref https://github.com/matrix-org/gomatrixserverlib/blob/a5277afd031f7bef7496704f67c074479fc6fe84/eventauth.go#L51
(Although I don't know which file "m.tag" should go to.)

Reply:
@anoadragon453 @babolivier Suggestions?

Reply:
The spot suggested by @APwhitehat looks good to me.